Add docs for GPU support (#660)
trixi-framework/TrixiParticles.jl: docs/src/overview.md

# Overview
The actual API reference is not listed on a single page, like in most Julia packages,
but instead is split into multiple sections that follow a similar structure
as the code files themselves.

> In these sections, API docs are integrated into theoretical background of the methods.

Reviewer: ```suggestion
In these sections, API docs are combined with explanations of the theoretical background of these methods.
```
